content/docs/cucumber/state.md

```diff
@@ -238,6 +238,211 @@ compile group: 'io.cucumber', name: 'cucumber-needle', version: '{{% version "cu
+
+```java
+package com.example.app.service.impl;
+
+import com.example.app.service.AppService;
+import com.google.inject.AbstractModule;
+
+public final class ServiceModule extends AbstractModule {
+    @Override
+    protected void configure() {
+        bind( AppService.class ).to( AppServiceImpl.class );
+        // ... (further bindings)
+    }
+}
+```
+
+The actual injector is then created like this: `injector = Guice.createInjector( new ServiceModule() );`
+
+This means we need to have access to the creation of the injector that Cucumber creates in order to customize
+it with our application specific modules.
+
+## The Cucumber object factory
+
+Whenever Cucumber needs a specific object, it asks an object factory for it.
+Cucumber has a default object factory that (in case of Guice) creates a default injector and
+delegates object creation to the injector.
+In case we want to customize this injector we need to provide our own object factory and tell Cucumber to use it.
+
+```java
+package com.example.app;
+
+import io.cucumber.core.backend.ObjectFactory;
+import io.cucumber.guice.CucumberModules;
+import io.cucumber.guice.ScenarioScope;
+import com.example.app.service.impl.ServiceModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Stage;
+
+public final class CustomObjectFactory implements ObjectFactory {
+
+    private Injector injector;
+
+    public CustomObjectFactory() {
+        // Create an injector with our service module
+        this.injector =
+          Guice.createInjector( Stage.PRODUCTION, CucumberModules.createScenarioModule(), new ServiceModule());
+    }
+
+    @Override
+    public boolean addClass( Class< ? > clazz ) {
+        return true;
+    }
+
+    @Override
+    public void start() {
+        this.injector.getInstance( ScenarioScope.class ).enterScope();
+    }
+
+    @Override
+    public void stop() {
+        this.injector.getInstance( ScenarioScope.class ).exitScope();
+    }
+
+    @Override
+    public < T > T getInstance( Class< T > clazz ) {
+        return this.injector.getInstance( clazz );
+    }
+}
+```
+
+This is almost the default object factory for Guice except that we have added our own bindings to the injector.
+
+Cucumber loads the object factory through the `java.util.ServiceLoader`. In order for the ServiceLoader to be able
+to pick up our custom implementation we need to provide the file `META-INF/services/io.cucumber.core.backend.ObjectFactory`.
+
+```
+com.example.app.CustomObjectFactory
+#
+# ... additional custom object factories could be added here
+#
+```
+
+Our step definition should now be working except that we have to tell Cucumber to use our custom object factory.
+There are several ways how this could be accomplished.
+
+### The command line
+
+When Cucumber is run from the command line, the custom object factory can be specified as class name argument.
```

Comment:
I like the conciseness in this particular section! Let's make the next sections as concise? (see my suggestions)